### ISSUE TITLE
anvi-gen-structure-database 

### ISSUE THREAD

**Issue by kevin-lab** [merenlab/anvio#1536]
Dear Anvio-Team,

anvi-gen-structure-database gives an Error.

The executable you requested is called `mod9.25`, but anvi'o doesn't agree with you that it is a working MODELLER program

## anvi'o version

```
Anvi'o version ...............................: esther (v6.2-master)
Profile DB version ...........................: 35
Contigs DB version ...........................: 19
Pan DB version ...............................: 14
Genome data storage version ..................: 7
Auxiliary data storage version ...............: 2
Structure DB version .........................: 2
Kegg Modules DB version ......................: 2
tRNA-seq DB version ..........................: 1

```

## System info

Linux 64 Bit
Ubuntu 18 LTS
8 GB RAM
Intel® Core™ i5 CPU M 520 @ 2.40GHz × 4 

## Description of Error

I tried to generate a structure database. I installed every needed third party program. Entered the license key in the correct position. And I'm receiving this issue.

Modeller is installed within anvio-master env.
Also DSSP.
```
The executable you requested is called `mod9.25`, but anvi'o doesn't agree with 
              you that it is a working MODELLER program. That was determined by running the   
              command `mod9.25 /tmp/tmpuc26jo3l/fasta_to_pir.py                               
              /home/kevin/github/anvio/anvio/tests/sandbox/mock_data_for_structure/proteins.fa
              /tmp/tmpuc26jo3l/test_out`, which raised the error seen above. If you want to   
              specify a specific MODELLER program, you can specify it with `--modeller-       
              executable`.
```
## Command used
```
anvi-gen-structure-database -c /home/kevin/Documents/GCA_009017715.1_ASM901771v1_genomic.reformatted.db  --gene-caller-ids 3976,2067 -o /media/kevin/DATA/GH99-STRUCTURE.db --pdb-db /media/kevin/DATA/PDB_database --modeller-executable mod9.25
```

Hopefully you can tell me what i do wrong.

Best,
Kevin

**Comment by ekiefl** [comment]
Hey @kevin-lab,

Please run this command again (with the flag `--debug`), and when the error pops up, try running the command suggested in the error message.

**Comment by kevin-lab** [comment]
Thank you for your very fast response and sorry to bother you again.

I still do not know what i should change now :(
I'm very sorry.


## Error Message

```
'import site' failed; use -v for traceback
Traceback (most recent call last):
  File "<string>", line 1, in ?
  File "/home/kevin/miniconda3/envs/anvio-master/lib/modeller-9.25/modlib/modeller/__init__.py", line 33, in ?
    _modeller.mod_license_key_set(config.license)
NameError: name '_modeller' is not defined

✖ anvi-gen-structure-database encountered an error after 0:00:03.193550

Traceback for debugging
================================================================================
  File "/home/kevin/github/anvio/bin/anvi-gen-structure-database", line 150, in <module>
    main(args)
  File "/home/kevin/github/anvio/anvio/terminal.py", line 841, in wrapper
    program_method(*args, **kwargs)
  File "/home/kevin/github/anvio/bin/anvi-gen-structure-database", line 36, in main
    structops.StructureSuperclass(args, create=True)._run()
  File "/home/kevin/github/anvio/anvio/structureops.py", line 381, in __init__
    self.sanity_check()
  File "/home/kevin/github/anvio/anvio/structureops.py", line 444, in sanity_check
    self.args.modeller_executable = MODELLER.check_MODELLER(self.modeller_executable)
  File "/home/kevin/github/anvio/anvio/drivers/MODELLER.py", line 920, in check_MODELLER
    "`--modeller-executable`." % (executable, " ".join(command)))
================================================================================


Config Error: The executable you requested is called `mod9.25`, but anvi'o doesn't agree with 
              you that it is a working MODELLER program. That was determined by running the   
              command `mod9.25 /tmp/tmp_z75r2e0/fasta_to_pir.py                               
              /home/kevin/github/anvio/anvio/tests/sandbox/mock_data_for_structure/proteins.fa
              /tmp/tmp_z75r2e0/test_out`, which raised the error seen above. If you want to   
              specify a specific MODELLER program, you can specify it with `--modeller-       
              executable`. 
```

Best,
Kevin

**Comment by ekiefl** [comment]
I'm confused. Did you run the command:

`mod9.25 /tmp/tmp_z75r2e0/fasta_to_pir.py /home/kevin/github/anvio/anvio/tests/sandbox mock_data_for_structure/proteins.fa /tmp/tmp_z75r2e0/test_out`

? As the error message suggests, anvio doesn't like your modeler program. By running the above command, maybe you can get some insight as to why. Perhaps you can try installing a different modeler version.

**Comment by kevin-lab** [comment]
Oh sorry. I am still a newbie to this.

## Command 

```mod9.25 /tmp/tmp_z75r2e0/fasta_to_pir.py /home/kevin/github/anvio/anvio/tests/sandbox mock_data_for_structure/proteins.fa /tmp/tmp_z75r2e0/test_out ```

## Error message 
```
Could not find platform independent libraries <prefix>
Could not find platform dependent libraries <exec_prefix>
Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]
'import site' failed; use -v for traceback
Traceback (most recent call last):
  File "<string>", line 1, in ?
  File "/home/kevin/miniconda3/envs/anvio-master/lib/modeller-9.25/modlib/modeller/__init__.py", line 31, in ?
    import _modeller
ImportError: No module named _modeller
```

**Comment by ekiefl** [comment]
No problem.

OK we are making progress.

So it seems your modeller installation did not work. My suggestion is to try installing modeller again. Try following instructions at salilab.org/modeller

**Comment by kevin-lab** [comment]
The fresh installation did not work either. But after a closer look on their website i saw this.

```Standard Python libraries like os aren't available or I get warnings such as ''import site' failed' or 'Could not find platform independent libraries <prefix>'
The mod9.25 script uses a built-in copy of the Python 2.3 interpreter, but not the accompanying modules. To fix this problem, you could install Python 2.3 on your system (however, you only need the Python 2.3 modules if you are trying to import non-Modeller modules; you don't need them to run regular Modeller scripts, so you can ignore these warnings). Alternatively, don't use the mod9.25 script to run Modeller, but see the instructions above for running Modeller with Python.```

Is this the problem i currently face?

How can i make it running?

Best,
Kevin

**Comment by ekiefl** [comment]
Hi Kevin, my suggestion is to contact the modeller developers. Determine through their tutorials how you can reliably test whether your installation has worked. Perhaps you could try a previous version of modeller. After you can successfully run a modeller command from their tutorials, I have faith that `anvi-gen-structure-database` will work.

I am closing this issue for now, since it is currently not an anvi'o issue, but a modeller issue. Sorry you are experiencing troubles, and good luck!
